### PR TITLE
[CWS] Add the ability to expose the value of some environment variables

### PR DESCRIPTION
**Pull request by YoannGh**
### What does this PR do?

This PR adds a new config option `runtime_security_config.envs_with_value` to pass a list of environment variable names that won't be concealed by the runtime security module.

Associated functional test:
`inv -e security-agent.functional-tests --testflags "-test.v -test.run TestProcessEnvsWithValue"`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -1094,6 +1094,7 @@ func InitConfig(config Config) {
 	bindEnvAndSetLogsConfigKeys(config, "runtime_security_config.activity_dump.remote_storage.endpoints.")
 	config.BindEnvAndSetDefault("runtime_security_config.event_stream.use_ring_buffer", false)
 	config.BindEnv("runtime_security_config.event_stream.buffer_size")
+	config.BindEnvAndSetDefault("runtime_security_config.envs_whitelist", []string{})
```
Comment (paulcacheux) [inline pkg/config/config.go:1097]: Maybe we could provide a nice set of defaults (like `LD_PRELOAD` etc)

**Review by paulcacheux — APPROVED** [review]

File: pkg/config/config.go (hunk 1 above)
Comment (YoannGh) [reply pkg/config/config.go:1097]: I'm fine adding default values like `LD_PRELOAD`, but shouldn't we add them directly to the config file for clarity

Comment (safchain) [reply pkg/config/config.go:1097]: I would rather use `runtime_security_config.envs_with_value`

Comment (safchain) [reply pkg/config/config.go:1097]: We should maybe update the config template file like for the sensitive words https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml#L1565-L1575

**Comment by paulcacheux** [comment]
Should we scrub the value ?

File: pkg/config/config.go (hunk 1 above)
Comment (YoannGh) [reply pkg/config/config.go]: Done

**Review by cswatt — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]
LGTM for @DataDog/agent-shared-components owned files

**Review by safchain — APPROVED** [review]

**Pull request merged**